src/containers/Tasks/ImportAttestation/ImportAttestation.tsx

```diff
@@ -1,78 +1,139 @@
 import * as sdk from '@kiltprotocol/sdk-js'
-import React from 'react'
-import { connect } from 'react-redux'
+import React, { useState, useEffect, useCallback } from 'react'
+import { connect, MapStateToProps } from 'react-redux'
 import AttestedClaimsListView from '../../../components/AttestedClaimsListView/AttestedClaimsListView'
 import ClaimDetailView from '../../../components/ClaimDetailView/ClaimDetailView'
-
+import PersistentStore, {
+  State as ReduxState,
+} from '../../../state/PersistentStore'
 import { notifySuccess } from '../../../services/FeedbackService'
 import * as Claims from '../../../state/ducks/Claims'
 
+type StateProps = {
+  claims: Claims.Entry[]
+}
+
 type DispatchProps = {
-  addAttestationToClaim: (attestation: sdk.IAttestedClaim) => void
+  addAttestedClaimToClaim: (attestedClaim: sdk.IAttestedClaim) => void
+  removeRequestForAttestation: (
+    claimId: Claims.Entry['id'],
+    rootHash: sdk.IRequestForAttestation['rootHash']
+  ) => void
 }
 
 type OwnProps = {
-  attestedClaim: sdk.IAttestedClaim
-
+  attestation: sdk.IAttestation
   onCancel?: () => void
   onFinished?: () => void
 }
 
-type Props = DispatchProps & OwnProps
+type Props = DispatchProps & OwnProps & StateProps
 
-class ImportAttestation extends React.Component<Props> {
-  constructor(props: Props) {
-    super(props)
-    this.state = {}
+const ImportAttestation: React.FC<Props> = ({
+  claims,
+  attestation,
+  onCancel,
+  onFinished,
+  addAttestedClaimToClaim,
+  removeRequestForAttestation,
+}) => {
+  const [requestForAttestationEntry, setRequestForAttestation] = useState<
+    sdk.IRequestForAttestation
+  >()
+  const [claimId, setClaimId] = useState<Claims.Entry['id']>()
 
-    this.onCancel = this.onCancel.bind(this)
-    this.importAttestation = this.importAttestation.bind(this)
-  }
+  const requestForAttest = useCallback((): void => {
```

Comment:
Instead of `useCallback`, you can just move the function contents into useEffect